### PR TITLE
DLS-7120 Upgrade to hmrc-mongo (and update all other libraries)

### PR DESCRIPTION
**Pull request by LAJW**
[Ticket](https://jira.tools.tax.service.gov.uk/browse/DLS-7120)

This project was so ancient I had to update _all_ libraries (SBT refused to download some of the dependencies).

 - Upgrade from `mockito-core` to `mockito-scala-specs2` since it's split into a Java-based core project and a Scala-based integration project.
 - Upgrade to sbt 1.7.2
 - Upgrade to scala 2.12.17
 - Disable warning suppression, dependency overrides
 - Remove `sbt-git-versioning`, `sbt-artifactory` and `sbt-service-manager` plugins which aren't used in other projects and were causing problems when building

### CODE REVIEW

**Comment by johno1985** [comment]
Looks like the build failed.

File: build.sbt [hunk 1]
```diff
@@ -33,21 +33,20 @@ dependencyOverrides += "com.typesafe.akka" %% "akka-http-core" % akkaHttpVersion
...
-  hmrc %% "reactivemongo-test" % "4.21.0-play-26" % scope,
-  hmrc %% "service-integration-test" % "0.12.0-play-26" % scope,
-  "org.scalatest" %% "scalatest" % "3.0.8" % scope,
-  "org.scalatestplus.play" %% "scalatestplus-play" % "3.1.2" %  scope,
+  hmrc %% "service-integration-test" % "1.3.0-play-28" % scope,
+  "org.scalatest" %% "scalatest" % "3.2.15" % scope,
+  "org.scalatestplus.play" %% "scalatestplus-play" % "5.1.0" %  scope,
   "org.pegdown" % "pegdown" % "1.6.0" % scope,
-  "org.mockito" % "mockito-core" % "3.5.10" % scope,
+  "org.mockito" % "mockito-core" % "5.1.1" % scope,
```
Comment (johno1985) [inline build.sbt]: I wonder if could use the mockito-scala library?

    "org.mockito"             %% "mockito-scala"              % "1.17.7"                % Test

File: project/plugins.sbt [hunk 2]
```diff
@@ -3,10 +3,8 @@ resolvers += Resolver.typesafeRepo("releases")
...
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.3.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.9.0")
 
+addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.2.0")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.1.0")
-
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.7" exclude("org.slf4j", "slf4j-simple"))
-
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.16" exclude("org.slf4j", "slf4j-simple"))
```
Comment (johno1985) [inline project/plugins.sbt:10]: do we still need the slfj exclusion?

**Comment by johno1985** [comment]
happy to approve once PR build has been fixed

File: build.sbt (hunk 1 above)
Comment (LAJW) [reply build.sbt]: We can and I did in the end `mockito-scala-specs2`

Comment (johno1985) [reply build.sbt]: Are we using both specs2 and scalatest? 

If we add the dependency as just 'mockito-scala' then mockito scala will grab the correct artifact for us, see https://github.com/mockito/mockito-scala#please-ensure-you-dont-declare-mockito-core-as-a-dependency-mockito-scala-will-pull-the-appropiate-version-automatically

File: project/plugins.sbt (hunk 2 above)
Comment (LAJW) [reply project/plugins.sbt:10]: Unclear. But we always have it in all repositories, and I'm too scared to remove it.

File: build.sbt (hunk 1 above)
Comment (LAJW) [reply build.sbt]: Fixed

**Comment by LAJW** [comment]
- Added flexmark as a dependency
 - Switched error handling (the new mongo connector no longer throws "duplicate error" keys)

**Pull request merged**